src/pages/clients/edit/components/AdditionalInfo.tsx

```diff
@@ -61,13 +60,43 @@ export function AdditionalInfo(props: Props) {
   const { id } = useParams();
 
   const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+
     const client = { ...props.client };
 
     setErrors(undefined);
 
     setClient(set(client as Client, event.target.id, event.target.value));
   };
 
+  const handleChangeSettings = (event: ChangeEvent<HTMLInputElement>) => {
+        
+    if (event.target.value.length === 0) {
+      const settingsProp = event.target.id.replace('settings.', '')
+
+      const _client = cloneDeep(props.client)
+      
+      if(_client){
+        delete _client.settings[settingsProp]
+        props.setClient(_client)
+        return
+      }
+    
+    }
+
+    const client = { ...props.client };
+
+    setErrors(undefined);
+
+    let value: any = event.target.value;
+
+    if (event.target.id === 'settings.send_reminders'){
+      value = event.target.value === 'enabled' ? true : false
+    }
+
+    setClient(set(client as Client, event.target.id, value));
```

Comment:
We can do better here.

I'd suggest following signature for this method:

```ts
const handleSettingsChange = (property: keyof Company['settings'], value: string | number) =>
```

This way we are not dealing with event directly and we can use `onValueChange` on select field.

Reply:
We can push this even further with type safety. Here's demo: https://tsplay.dev/mxJDKw

Reply:
This still pushes the type checks into the method now that value is either string/number.

In settings the concept of a blank string unsets the prop from the object. but now that value is either string/number we cannot use .length so it does complicate the logic further down?